### ISSUE TITLE
Unhelpful error when using cabal.project.local without cabal.project

### ISSUE THREAD

**Issue by brandonchinn178** [haskell/cabal#9168]
**Describe the bug**
If `cabal.project.local` exists without `cabal.project`, Cabal complains with:
```
Warning: both implicit and explicit configuration is present.When using configuration(s) from /Users/bchinn/Desktop/foo/cabal.project.local, the following errors occurred:
The package location glob './*.cabal' does not match any files or directories.
```

This happens even if `cabal.project.local` contains a `packages:` line.

Also, it seems like a newline is missing after the warning message?

**To Reproduce**
Steps to reproduce the behavior: (fails = shows bug, succeeds = expectedly fails because cabal file is empty)

```bash
mkdir foo
touch foo/foo.cabal
cat 'packages: foo/' >> cabal.project.local

# fails
cabal build

# succeeds with just an empty cabal.project
touch cabal.project
cabal build
```

**Expected behavior**
Renaming `cabal.project` to `cabal.project.local` should have the same behavior.

**System information**
 - Mac
 - `cabal`: 3.10.1.0

**Additional context**

**Comment by andreabedini** [comment]
I agree the message is not very helpful but there are reasons:tm: behind it.

To configure a project one uses the `cabal configure` command. This will take v2/nix-style/project-config options and write them in `cabal.project.local`. This works because `packages: .` is implied by the missing `cabal.project`.

One assumption is that we use `cabal configure` to make ad-hoc, local changes to a cabal project (e.g. _I need to use this compiler ... `cabal configure -w ghc-9.4.5`_). Permantent changes would be done by editing the existing cabal.project or by creating one.

My understanding is that cabal is concerned about "merging" the implicit configuration (`packages: .`) with the one in `cabal.project.local` (Note: `projectPackages` is a list so merging is concatenation).

We can discuss if this is something to worry about or not; off the top of my head I cannot think of any bad consequences.

Just to understand, how did you get to add `packages:` to `cabal.project.local`? why not `cabal.project`?

**Comment by brandonchinn178** [comment]
I was contributing to the head.hackage project, and they don't commit a cabal.project file, but they have a script that generates cabal.project.local (without cabal configure).

I guess their script could be updated to generate cabal.project + gitignore cabal.project.

**Comment by andreabedini** [comment]
> I was contributing to the head.hackage project, and they don't commit a cabal.project file, but they have a script that generates cabal.project.local (without cabal configure).

I think I understand now. Did you need to add an extra package? Was it something from a Hackage (or other repository), or it was something you were just writing?
Thank you <3!

**Comment by gbaz** [comment]
I would suggest that the easiest path is that they _do_ commit a cabal.project file, and it it contains the "packages" stanza and nothing else, and then their script generates a local with the appropriate local stuff.

**Comment by brandonchinn178** [comment]
@andreabedini I needed to add a patch for a package on Hackage to work with GHC HEAD. The script that head.hackage provides unpacks the source code + adds it as an optional package to a cabal.project.local file.

@gbaz Yes, that's what I did, but I think the error message should still be improved. It took me quite a while to figure out what was happening
